apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -60,26 +73,36 @@ def _guess_location(item: pystac.Item) -> Tuple[str, bool]:
     return self_link, relative
 
 
-def get_items(
-    search: ItemSearch
-) -> Generator[Tuple[dict, str, bool], None, None]:
-    items = search.get_all_items()
+def item_to_meta_uri(item: Item) -> Generator[Tuple[dict, str, bool], None, None]:
+    uri, relative = _guess_location(item)
+    try:
+        metadata = item.to_dict()
+        if relative:
+            metadata = stac_transform(metadata)
+        else:
+            metadata = stac_transform_absolute(metadata)
+    except KeyError as e:
+        logging.error(f"Failed to handle item with KeyError: '{e}'\n The URI was {uri}")
```

Comment:
This exception handling looks wrong, it looks like it'll just blindly continue after logging an error.

Reply:
You're right. Fixed.